### PR TITLE
Add dotenv

### PR DESCRIPTION
**Pull request by p4veI**
I was thinking if we could use dotenv loader for the dev/test run setup, I know I can set env vars in IDE but I consider this better.

closes https://github.com/staabm/phpstan-dba/issues/267

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
 		"ext-mysqli": "*",
 		"composer-runtime-api": "^2.0",
 		"composer/semver": "^3.2",
-		"phpstan/phpstan": "^1.2"
+		"phpstan/phpstan": "^1.2",
+		"vlucas/phpdotenv": "^5.4"
```
Comment (staabm) [inline composer.json:11]: I think we can move this to require-dev, as we only use it for the test-suite

**Comment by staabm** [comment]
running `composer test` with this PR I get:

```
$ composer test
> phpunit -c tests/default/config/phpunit.xml
PHPUnit 9.5.16 by Sebastian Bergmann and contributors.

Error in bootstrap script: Error:
mysqli object is already closed
PHP Warning:  Undefined array key "DBA_HOST" in C:\dvl\Workspace\phpstan-dba\tests\ReflectorFactory.php on line 29
PHP Warning:  Undefined array key "DBA_USER" in C:\dvl\Workspace\phpstan-dba\tests\ReflectorFactory.php on line 30
PHP Warning:  Undefined array key "DBA_PASSWORD" in C:\dvl\Workspace\phpstan-dba\tests\ReflectorFactory.php on line 31
PHP Warning:  Undefined array key "DBA_DATABASE" in C:\dvl\Workspace\phpstan-dba\tests\ReflectorFactory.php on line 32
PHP Warning:  Undefined array key "DBA_MODE" in C:\dvl\Workspace\phpstan-dba\tests\ReflectorFactory.php on line 33
PHP Warning:  Undefined array key "DBA_REFLECTOR" in C:\dvl\Workspace\phpstan-dba\tests\ReflectorFactory.php on line 34
PHP Warning:  mysqli::__construct(): (HY000/1045): Access denied for user ''@'192.168.50.51' (using password: NO) in C:\dvl\Workspace\phpstan-dba\tests\ReflectorFactory.php on line 54
Script phpunit -c tests/default/config/phpunit.xml handling the phpunit event returned with error code 1
Script @phpunit was called via test
```

did you intentionally not put the defaults we had before into the .dev.dist file?

**Comment by p4veI** [comment]
@staabm sort of, I originally wanted to keep the defaults in the reflector factory, so it's only required for other connections and modes, but it I could use the default to null with default postgres instances..

maybe I'll default everything to null so it's more clear the defined env is required...

Probably could add this to library docs as well and describe the dev setup...

**Comment by staabm** [comment]
Whether the consumer of phpstan-dba uses dotenv (or a different impl) or no dotenv at all is not subject of phpstan-dba.


Its the choice of the user

**Comment by p4veI** [comment]
> 

I've cleaned up the setup - now if you don't have IDE env vars setup it should default to what you have in `.env`, hope that's what you were suggesting.

File: tests/default/config/bootstrap.php [hunk 2]
```diff
@@ -1,11 +1,17 @@
 <?php
 
+use Dotenv\Dotenv;
 use staabm\PHPStanDba\QueryReflection\QueryReflection;
 use staabm\PHPStanDba\QueryReflection\RuntimeConfiguration;
 use staabm\PHPStanDba\Tests\ReflectorFactory;
 
 require_once __DIR__.'/../../../vendor/autoload.php';
 
+if (false === getenv('GITHUB_ACTION')) {
```
Comment (staabm) [inline tests/default/config/bootstrap.php:10]: I don't think we need this IF-case arround the loading.

another question: does the dotenv class automatically load the .env.dist file now?
I tried running this PR locally and I am still getting errors
```
$ composer test
> phpunit -c tests/default/config/phpunit.xml
PHPUnit 9.5.16 by Sebastian Bergmann and contributors.

Error in bootstrap script: RuntimeException:
Unknown mode:
PHP Warning:  Undefined array key "DBA_HOST" in C:\dvl\Workspace\phpstan-dba\tests\ReflectorFactory.php on line 29
PHP Warning:  Undefined array key "DBA_USER" in C:\dvl\Workspace\phpstan-dba\tests\ReflectorFactory.php on line 30
PHP Warning:  Undefined array key "DBA_PASSWORD" in C:\dvl\Workspace\phpstan-dba\tests\ReflectorFactory.php on line 31
PHP Warning:  Undefined array key "DBA_DATABASE" in C:\dvl\Workspace\phpstan-dba\tests\ReflectorFactory.php on line 32
PHP Warning:  Undefined array key "DBA_MODE" in C:\dvl\Workspace\phpstan-dba\tests\ReflectorFactory.php on line 33
PHP Warning:  Undefined array key "DBA_REFLECTOR" in C:\dvl\Workspace\phpstan-dba\tests\ReflectorFactory.php on line 34
```

Comment (staabm) [reply tests/default/config/bootstrap.php:10]: ahh I get it now. I need to copy the .env.dist file to .env once.. it works now

Comment (p4veI) [reply tests/default/config/bootstrap.php:10]: yeah, the .dist file is supposed to be like a template for the envs..

But it's commonly used so some defaults / test setup can be blaced in dist and setup from the existing file..

Comment (staabm) [reply tests/default/config/bootstrap.php:10]: I see. I am used to a process where the .dist file is automtically picked up, without the need to copy it a priori.

but thats fine.

thanks for the work you put into it here. I will merge it.

File: bootstrap.php [hunk 3]
```diff
@@ -1,11 +1,17 @@
 <?php
 
+use Dotenv\Dotenv;
 use staabm\PHPStanDba\QueryReflection\QueryReflection;
 use staabm\PHPStanDba\QueryReflection\RuntimeConfiguration;
 use staabm\PHPStanDba\Tests\ReflectorFactory;
 
 require_once __DIR__.'/vendor/autoload.php';
 
+if (false === getenv('GITHUB_ACTION')) {
```
Comment (p4veI) [inline bootstrap.php:10]: just wanted to add to the rest of bootstraps.. the if might be extra

**Comment by staabm** [comment]
thank you

File: bootstrap.php (hunk 3 above)
Comment (staabm) [reply bootstrap.php:10]: did you mean there is something missing?

Comment (p4veI) [reply bootstrap.php:10]: it's fine))

**Pull request merged**